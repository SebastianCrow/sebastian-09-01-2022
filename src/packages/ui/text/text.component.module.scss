@import "../../../styles/themes/dark";

.primary {
  color: $color;
}

.secondary {
  color: $color-secondary;
}

.success {
  color: $color-success;
}

.danger {
  color: $color-danger;
}
