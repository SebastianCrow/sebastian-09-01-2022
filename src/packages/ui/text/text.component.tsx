import React, { FunctionComponent } from 'react';
import cns from 'classnames';
import styles from './text.component.module.scss';

export type TextVariant = 'regular' | 'title' | 'code';

export type TextColor = 'primary' | 'secondary' | 'success' | 'danger';

export interface TextProps {
  variant?: TextVariant;
  color?: TextColor;
}

export const Text: FunctionComponent<TextProps> = ({
  variant,
  color = 'primary',
  children,
}) => {
  return (
    <div
      className={cns({
        [styles.title]: variant === 'title',
        [styles.code]: variant === 'code',
        [styles.primary]: color === 'primary',
        [styles.secondary]: color === 'secondary',
        [styles.success]: color === 'success',
        [styles.danger]: color === 'danger',
      })}
    >
      {children}
    </div>
  );
};
