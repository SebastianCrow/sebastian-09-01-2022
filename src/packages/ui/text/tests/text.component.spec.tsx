import React from 'react';
import { Text } from '../text.component';
import { create } from 'react-test-renderer';

describe('text.component', () => {
  test('renders', () => {
    const renderer = create(<Text>Test</Text>);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
