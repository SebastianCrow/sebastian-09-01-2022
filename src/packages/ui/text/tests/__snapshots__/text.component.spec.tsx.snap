// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text.component renders 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 primary css-ahj2mt-MuiTypography-root"
>
  Test
</p>
`;
